src/LiteVault.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./interfaces/IAuthorize.sol";
+import "./interfaces/IVault.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+
+/**
+ * @title LiteVault
+ * @notice A simple vault that allows users to deposit and withdraw tokens.
+ */
+contract LiteVault is IVault, ReentrancyGuard {
+    // Mapping from user to token to balances
+    mapping(address => mapping(address => uint256)) private balances;
+
+    IAuthorize public authorizer;
+    address public owner;
+
+    event Deposit(address indexed user, address indexed token, uint256 amount);
+    event Withdrawal(address indexed user, address indexed token, uint256 amount);
+
+    /**
+     * @dev Modifier to check if the caller is the owner of the contract.
+     */
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Not the contract owner");
+        _;
+    }
```

Comment:
Todo, we need to add the ability to transfer Ownership,
Either we copy Ownable.sol or implement here